### PR TITLE
Implement camera handling APIs in objective-c

### LINKED ISSUE #849: Map camera position API
setters and getters for camera position

### PR DESCRIPTION
**Pull request by msmollin**
Includes tilt, rotation, zoom, and position management, as well as camera type switching. Fixes #849, #850, #851, #858, #897

### CODE REVIEW

File: ios/src/TGMapViewController.mm [hunk 1]
```diff
@@ -1,101 +1,222 @@
...
-    self.renderRequested = true;
-    self.continuous = false;
-
     if (!self.context) {
         NSLog(@"Failed to create ES context");
     }
-
+    self.pixelScale = [[UIScreen mainScreen] scale];
+    self.renderRequested = YES;
+    self.continuous = YES;
```
Comment (karimnaaji) [inline ios/src/TGMapViewController.mm]: For consistency with android we should leave continuous rendering to `NO`.

File: ios/src/TGMapViewController.mm [hunk 2]
```diff
@@ -1,101 +1,222 @@
...
+            return Tangram::EaseType::quint;
+        case TangramEaseTypeCubic:
+            return Tangram::EaseType::cubic;
+        default:
+            return Tangram::EaseType::quint;
+    }
+}
+
+- (void)setZoomEased:(float)zoomLevel withDuration:(float)duration {
+    [self setZoomEased:zoomLevel withDuration:duration withEaseType:TangramEaseTypeQuint];
```
Comment (karimnaaji) [inline ios/src/TGMapViewController.mm]: Default for ease is `Cubic` on android: https://github.com/tangrams/tangram-es/blob/master/android/tangram/src/com/mapzen/tangram/MapController.java#L51

Comment (blair1618) [inline ios/src/TGMapViewController.mm]: Though it's 'quint' in the C++ interface. Maybe it should be linear everywhere or just not have a default, I'm not sure what I was thinking when I made these defaults.

Comment (karimnaaji) [inline ios/src/TGMapViewController.mm]: Either works, it's seems ok to leave defaults for ease (removing the default would involve a method deletion on the [API](https://github.com/tangrams/tangram-es/blob/master/android/tangram/src/com/mapzen/tangram/MapController.java#L238_L257)). we should just have them consistent across platforms.

Comment (msmollin) [inline ios/src/TGMapViewController.mm]: Sooooo linear or cubic? :) I agree with Karim that we should have consistency. I can change to cubic if that's the consensus.

Comment (blair1618) [inline ios/src/TGMapViewController.mm]: Let's just go with cubic for consistency with Android!

**Comment by msmollin** [comment]
Closing this PR and re-opening against master.

**Pull request closed without merging**